.github/workflows/npmpublish.yml

```diff
@@ -3,31 +3,30 @@ on:
   push:
     branches:
       - main
+  pull_request:
+    branches: [main]
 
 jobs:
-  publish:
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v3.1.0
-      - uses: ./.github/actions/pnpm-setup-action
-      - run: pnpm -r publish --access public
-        env:
-          NODE_AUTH_TOKEN: ${{secrets.NPM_TOKEN}}
-  tag:
+  tag-and-publish:
     strategy:
       matrix:
         publishablePackage: [synapse-react-client, markdown-it-synapse]
     runs-on: ubuntu-latest
     steps:
-      # Now check if the package version changed
+      - uses: actions/checkout@v3.1.0
+      - uses: ./.github/actions/pnpm-setup-action
+      # Check if the package version changed
       - id: package-version-check
         uses: EndBug/version-check@v2
         with:
-          file-name: ./packages/${{ matrix.publishablePackage }}/package.json
-      # If the package version changed, create the tag using the output of the previous step.
+          file-name: './packages/${{ matrix.publishablePackage }}/package.json'
+      # Build the package to publish
+      - name: Build package
+        run: pnpm nx run ${{ matrix.publishablePackage }}:build
+      # If the package version changed, and the build succeeded, create the tag using the output of the previous step.
```

Comment:
Adding this step so if the build fails, the job will exit without creating a tag or publishing